fastapi/encoders.py

```diff
@@ -142,6 +105,54 @@ def jsonable_encoder(
             )
         return encoded_list
 
+    def encode_BaseModel(obj: BaseModel) -> Any:
+        encoder = getattr(obj.__config__, "json_encoders", {})
+        if custom_encoder:
+            encoder.update(custom_encoder)
+
+        obj_dict = obj.dict(
+            include=include,
+            exclude=exclude,
+            by_alias=by_alias,
+            exclude_unset=exclude_unset,
+            exclude_none=exclude_none,
+            exclude_defaults=exclude_defaults,
+        )
+        if "__root__" in obj_dict:
+            obj_dict = obj_dict["__root__"]
+
+        return jsonable_encoder(
+            obj_dict,
+            exclude_none=exclude_none,
+            exclude_defaults=exclude_defaults,
+            custom_encoder=encoder,
+            sqlalchemy_safe=sqlalchemy_safe,
+        )
+
+    # Use type comparsions on common types before expensive isinstance checks
+    if type(obj) in (str, int, float, type(None)):
```

Comment:
Profiling with and without obj_type made a negligable difference, I imagine that's because getting an object's type is such a simple operation for the interpreter to perform.